src/pad_controller.rs

```diff
@@ -7,29 +7,28 @@ use glib::IsA;
 use gtk_sys;
 use PadActionEntry;
 use PadController;
+use libc::c_char;
 
 pub trait PadControllerExtManual: 'static {
-    #[cfg(any(feature = "v3_22", feature = "dox"))]
     fn set_action_entries(&self, entries: &[PadActionEntry]);
 }
 
 impl<O: IsA<PadController>> PadControllerExtManual for O {
-    #[cfg(any(feature = "v3_22", feature = "dox"))]
     fn set_action_entries(&self, entries: &[PadActionEntry]) {
         let n_entries = entries.len() as i32;
         let entry_strings = entries
             .iter()
-            .map(|e| (CString::new(&e.label), CString::new(&e.action_name)))
-            .collect::<Vec<_>>();
+            .map(|e| (e.label.to_glib_none().0, e.action_name.to_glib_none().0))
+            .collect::<Vec<(*mut c_char, *mut c_char)>>();
```

Comment:
Now if  dangling pointers.
First vector must be owned or stashed `.to_glib_none()` without `.0`

Reply:
* Now we have  dangling pointers again

Reply:
Ok, kept the `Stashed` instead.